configs/example/riscv-se.py

```diff
@@ -0,0 +1,71 @@
+import argparse
+
+from m5.objects import RiscvO3CPU
+
+from gem5.components.boards.simple_board import SimpleBoard
+from gem5.components.cachehierarchies.classic.private_l1_private_l2_cache_hierarchy import (
+    PrivateL1PrivateL2CacheHierarchy,
+)
+from gem5.components.memory import SingleChannelDDR3_1600
+from gem5.components.processors.base_cpu_core import BaseCPUCore
+from gem5.components.processors.base_cpu_processor import BaseCPUProcessor
+from gem5.isas import ISA
+from gem5.resources.resource import BinaryResource
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+
+parser = argparse.ArgumentParser()
+parser.add_argument("binary", type=str)
+parser.add_argument("arguments", nargs="*", type=str, default=[])
+parser.add_argument("-c", "--cores", required=False, type=int, default=1)
+parser.add_argument("-v", "--vlen", required=False, type=int, default=256)
+parser.add_argument("-e", "--elen", required=False, type=int, default=64)
+
+args = parser.parse_args()
+
+requires(isa_required=ISA.RISCV)
+
+cache_hierarchy = PrivateL1PrivateL2CacheHierarchy(
+    l1d_size="32KiB", l1i_size="32KiB", l2_size="512KiB"
+)
+
+# Setup the system memory.
+memory = SingleChannelDDR3_1600()
+
+
+def create_cores(num_cores, vlen, elen):
+    cores = [
+        BaseCPUCore(
+            core=RiscvO3CPU(cpu_id=i),
+            isa=ISA.RISCV,
+        )
+        for i in range(num_cores)
+    ]
+
+    for i in range(num_cores):
+        for j in range(cores[i].core.numThreads):
+            cores[i].core.isa[j].vlen = vlen
+            cores[i].core.isa[j].elen = elen
+
+    return cores
```

Comment:
Instead of this, can you extend the `BaseCPUCore`? We would prefer to show people this style as "examples" since it's more extensible if you have more customizations.

E.g.,

```python
class RVVCore(BaseCPUCore):
    def __init__(self, vlen, elen):
        super.__init__(core=RiscvO3CPU(cpu_id=i), isa=ISA.RISCV)
        self.core.isa.vlen = vlen
        self.core.isa.elen = elen
```